packages/vm/src/errors.rs

```diff
@@ -1,72 +1,85 @@
-use std::fmt::Debug;
-use std::io;
+use std::fmt::{Debug, Display};
 
 use snafu::Snafu;
-use wasmer_runtime_core::error as core_error;
 
 #[derive(Debug, Snafu)]
-#[snafu(visibility = "pub")]
+#[non_exhaustive]
 pub enum VmError {
+    /// An error coming from the backend (i.e. the chain)
+    #[snafu(display("Backend error: {}", msg))]
+    BackendErr {
+        msg: String,
+        backtrace: snafu::Backtrace,
```

Comment:
Do we want to do the same `Option<Backtrace>` thing as with `StdError`?
Or was that just to enable serialization and we don't are about serialization here (yet)

Reply:
Option was required since in cotrast to `Backtrace`, `Option<Backtrace>` has a default value, which is needed for deserialization. I don't think we need deserialization here right now.